integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/fieldtype/ElasticsearchDecimalScaleIT.java

```diff
@@ -0,0 +1,210 @@
+				.setup();
+
+		IndexWorkPlan<? extends DocumentElement> workPlan = indexManager.createWorkPlan();
+		workPlan.add( referenceProvider( "1" ), doc -> doc.addValue( bothScaleIndexMapping.scaled, BigDecimal.valueOf( 739.11111 ) ) );
+		workPlan.execute().join();
+
+		// default decimal scale is 2
+		// decimal scale has been set to 3, overriding the default
+		// decimal scale affects search precision, since 739.11111 will be treated as 739.111
+		SearchQuery<DocumentReference> query = indexManager.createSearchScope()
+				.query().asReference()
+				.predicate( p -> p.range().onField( "scaled" ).above( BigDecimal.valueOf( 739.11 ) ).excludeLimit() )
+				.toQuery();
+		assertThat( query ).hasDocRefHitsExactOrder( INDEX_NAME, "1" );
+
+		query = indexManager.createSearchScope()
+				.query().asReference()
+				.predicate( p -> p.range().onField( "scaled" ).above( BigDecimal.valueOf( 739.111 ) ).excludeLimit() )
+				.toQuery();
+		assertThat( query ).hasNoHits();
+	}
+
+	@Test
+	public void decimalScale_doesNotAffectProjections() {
+		setupHelper.withDefaultConfiguration()
+				.withIndex( INDEX_NAME,
+						ctx -> this.bothScaleIndexMapping = new BothScaleIndexMapping( ctx ),
+						indexManager -> this.indexManager = indexManager )
+				.setup();
+
+		IndexWorkPlan<? extends DocumentElement> workPlan = indexManager.createWorkPlan();
+		workPlan.add( referenceProvider( "1" ), doc -> doc.addValue( bothScaleIndexMapping.scaled, BigDecimal.valueOf( 739.11111 ) ) );
+		workPlan.execute().join();
+
+		// default decimal scale is 2
+		// decimal scale has been set to 3, overriding the default
+		// decimal scale should not have any effects on stored valued for projections
+		SearchQuery<Object> query = indexManager.createSearchScope().query()
+				.asProjection( p -> p.field( "scaled" ) )
+				.predicate( p -> p.matchAll() )
+				.toQuery();
+		assertThat( query ).hasHitsExactOrder( BigDecimal.valueOf( 739.11111 ) );
+	}
+
+	@Test
+	public void decimalScale_doesNotAffectProjections_exceedingZeros() {
+		setupHelper.withDefaultConfiguration()
+				.withIndex( INDEX_NAME,
+						ctx -> this.bothScaleIndexMapping = new BothScaleIndexMapping( ctx ),
+						indexManager -> this.indexManager = indexManager )
+				.setup();
+
+		IndexWorkPlan<? extends DocumentElement> workPlan = indexManager.createWorkPlan();
+		workPlan.add( referenceProvider( "1" ), doc -> doc.addValue( bothScaleIndexMapping.scaled, BigDecimal.valueOf( 739.11111000 ) ) );
+		workPlan.execute().join();
+
+		// default decimal scale is 2
+		// decimal scale has been set to 3, overriding the default
+		// decimal scale should not have any effects on stored valued for projections
+		// any exceeding decimal value zero will be ignored
+		SearchQuery<Object> query = indexManager.createSearchScope().query()
+				.asProjection( p -> p.field( "scaled" ) )
+				.predicate( p -> p.matchAll() )
+				.toQuery();
+		assertThat( query ).hasHitsExactOrder( BigDecimal.valueOf( 739.11111 ) );
+	}
+
```

Comment:
One thing you forgot to test (and, I think, to implement): when targeting multiple indexes in a search query, there should be an error when the targeted field of a match or range query, or of a sort, is indexed with a different decimal scale in both indexes.

Maybe some of this will work in Elasticsearch; but I'm pretty sure it won't work in our Lucene backend.

Reply:
Can I add those tests to `DecimalScaleIT`?
Actually I would avoid to make `MatchSearchPredicateIT`, `RangeSearchPredicateIT`, `FieldSearchSortIT` more complex.
But as you prefer.

Reply:
I would like for these classes same as for `FieldSearchProjectionIT` make them using only the field created by the FieldTypeDescriptor(s). Moving the rest on different classes. What you think about it?

Reply:
> Can I add those tests to `DecimalScaleIT`?
> Actually I would avoid to make `MatchSearchPredicateIT`, `RangeSearchPredicateIT`, `FieldSearchSortIT` more complex.
> But as you prefer.

I really dislike the idea of testing a single feature (match, range, ...) in many different classes; to me it just makes it harder to see what is expected from that feature. Because in the end, it's not the decimal scale feature you're testing, it really is the field compatibility checks when building predicates.

I won't stop you, but I think it hurts more than it helps. If it's about making test classes smaller, splitting `MatchSearchPredicateIT` into multiple classes in the same package would be a better solution in my opinion (note I'm not suggesting to do it in this PR).

> I would like for these classes same as for FieldSearchProjectionIT make them using only the field created by the FieldTypeDescriptor(s).

I do not understand this sentence.

Reply:
Not a problem. 
About the last sentence. In those tests we have 2 kind of field:

1. Some generated by an instance of 'FieldTypeDescriptor', such as `supportedFieldModels`
2. Others declared explicitly. like 'whitespaceAnalyzedField'
I would like to move all fields of type 1 out of this test class.

My concern is about readability of the test itself. In my opinion is getting too complex.
But as you wish.

Reply:
> I would like to move the latter out of this test class.

`FieldSearchProjectionIT` is about field projections, while `SearchProjectionIT` is about projections in general. You can't apply the same reasoning to match predicates, as the match predicates always apply to a field.

You could have `MatchSearchPredicateBaseIT` for your item 2, and `MatchSearchPredicateTypeIT` for item 1. But then the "base" tests would miss some very basic tests... it would just be weird. There might be a way to split these classes, but it's not as obvious as you're suggesting it is.

If you want to reorganize tests, please open a ticket and state the problem so we can discuss it. Trying to sneak this kind of organizational change in a PR about something else entirely is bound to simply lead to an inconsistent codebase ;)